### ISSUE TITLE
Deisiany / BSgenome.Gmax.EnsemblPlants.Gmv2

### ISSUE THREAD

**Issue by Deisiany** [Bioconductor/Contributions#1752]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/Deisiany/BSgenome.Gmax.NCBI.Gmv40.git

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [ x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [ x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [ x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [ x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [ x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [ x] The 'devel' branch for new packages and features.
- [ x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [ x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/

**Comment by Deisiany** [comment]
I have already emailed the bioconductor team, but I want to warn here as well that I was not able to upload my 
"single_sequences.2bit" file because it is to big, so I would appreciate if there is any way that I could share that  file with you. Without it I believe the package wouldn’t work.

Thank you!

**Comment by mtmorgan** [comment]
Following conversation with @hpages , we'll close this issue and he will review and add the package manually.

**Comment by hpages** [comment]
Please see the support site for my answer to your question about this: https://support.bioconductor.org/p/p132627/#p132628

Thanks!

**Comment by Deisiany** [comment]
Hello,
Would like to follow up on this issue.
Is everything alright with the package submission?
Or anything that I can do to help?

Thank you!

**Comment by hpages** [comment]
Hi @Deisiany , I'll take a look at BSgenome.Gmax.EnsemblPlants.Gmv2 today.

H.

**Comment by Deisiany** [comment]
Thank you Herve.


Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by hpages** [comment]
Hi @Deisiany ,
I'm taking a look at the package. Please ignore the above message from the bioc-issue-bot, there's no need to trigger a new build (the build would fail anyways).
H.

**Comment by hpages** [comment]
Hmm.. I'm not sure it's a good idea to use this kind of names for the chromosomes:
```
> seqinfo(BSgenome.Gmax.EnsemblPlants.Gmv2)
Seqinfo object with 20 sequences from v2.1 genome:
  seqnames                                       seqlengths isCircular genome
  Glycine_max.Glycine_max_v2.1.dna.chromosome.1    56831624      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.2    48577505      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.3    45779781      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.4    52389146      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.5    42234498      FALSE   v2.1
  ...                                                   ...        ...    ...
  Glycine_max.Glycine_max_v2.1.dna.chromosome.16   37887014      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.17   41641366      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.18   58018742      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.19   50746916      FALSE   v2.1
  Glycine_max.Glycine_max_v2.1.dna.chromosome.20   47904181      FALSE   v2.1
```
If I create a TxDb object for this assembly I get:
```
library(GenomicFeatures)
gmax_txdb <- makeTxDbFromBiomart(biomart="plants_mart", dataset="gmax_eg_gene", host="https://plants.ensembl.org")
seqinfo(gmax_txdb)
# Seqinfo object with 1190 sequences from an unspecified genome:
#   seqnames seqlengths isCircular genome
#   1          56831624       <NA>   <NA>
#   2          48577505       <NA>   <NA>
#   3          45779781       <NA>   <NA>
#   4          52389146       <NA>   <NA>
#   5          42234498       <NA>   <NA>
#   ...             ...        ...    ...
#   KZ848298       3151       <NA>   <NA>
#   KZ848299       2711       <NA>   <NA>
#   KZ848300       3302       <NA>   <NA>
#   KZ848301       9068       <NA>   <NA>
#   KZ848302       2655       <NA>   <NA>
```
So I would suggest that you get rid of the `Glycine_max.Glycine_max_v2.1.dna.chromosome.` prefix.

Also the `genome` column should contain the name of the assembly i.e. `Gmv2` instead of `v2.1`. This is something that you used to be able to specify via the `provider_version:` field in the seed file but that field got replaced with the `genome` field in more recent versions of `BSgenome::forgeBSgenomeDataPkg()`.

Thanks,
H.

**Comment by Deisiany** [comment]
Hi Hervé,
I am working on it. However it seems a lot of things have changed and I
will probably need to work more on it.
For instance, now I am not being able to read my package because it sees to
no accept anything else that "is not a registered NCBI assembly or UCSC
genome"
and as you probably know, that is not listed in either
"registered_NCBI_assemblies()
or registered_UCSC_genomes() to list the NCBI or UCSC
 assemblies/genomes currently registered in the GenomeInfoDb package".
*How should I proceed?*

*Best,*

Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by Deisiany** [comment]
Hi @hpages,
I am trying to switch too UCSI instead to get the package out, however neither NCBI or UCSI seems to have Glycine max registered.

I have seen other issues that had the same problem and your response to it was register sequences.
could you please do the same for Glycine max for me?

Thank you!

**Comment by hpages** [comment]
Hi @Deisiany ,

Sorry for the slow response. I registered a few `Glycine max` NCBI assemblies in **GenomeInfoDb** 1.31.4:
```
library(GenomeInfoDb)

subset(registered_NCBI_assemblies(), organism=="Glycine max")
#       organism         assembly       date               extra_info
# 94 Glycine max Glycine_max_v2.1 2018/07/24     Cultivar:Williams 82
# 95 Glycine max Glycine_max_v4.0 2021/03/10     Cultivar:Williams 82
# 96 Glycine max     ASM2049715v1 2021/10/14       Cultivar:Hwangkeum
# 97 Glycine max   GmBRS537.a1.v2 2021/10/15 Cultivar:EMBRAPA BRS 537
#    assembly_accession circ_seqs
# 94    GCF_000004515.5  MT, Pltd
# 95    GCF_000004515.6  MT, Pltd
# 96    GCA_020497155.1          
# 97    GCA_012273815.2     

Seqinfo(genome="Glycine_max_v2.1")
# Seqinfo object with 1192 sequences (2 circular) from Glycine_max_v2.1 genome:
#   seqnames      seqlengths isCircular           genome
#   Chr01           56831624      FALSE Glycine_max_v2.1
#   Chr02           48577505      FALSE Glycine_max_v2.1
#   Chr03           45779781      FALSE Glycine_max_v2.1
#   Chr04           52389146      FALSE Glycine_max_v2.1
#   Chr05           42234498      FALSE Glycine_max_v2.1
#   ...                  ...        ...              ...
#   scaffold_2573       3151      FALSE Glycine_max_v2.1
#   scaffold_2654       2711      FALSE Glycine_max_v2.1
#   scaffold_2706       3302      FALSE Glycine_max_v2.1
#   scaffold_2709       9068      FALSE Glycine_max_v2.1
#   scaffold_2735       2655      FALSE Glycine_max_v2.1
```

To forge **BSgenome.Gmax.NCBI.Gmv21** I strongly recommend that you use a `.2bit` file instead of a FASTA file. Converting a FASTA file to the `.2bit` format is not hard. You could download `GCF_000004515.5_Glycine_max_v2.1_genomic.fna.gz` from https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/515/GCF_000004515.5_Glycine_max_v2.1/, or `GCA_000004515.4_Glycine_max_v2.1_genomic.fna.gz` from https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/004/515/GCA_000004515.4_Glycine_max_v2.1/. Your choice. Both files should contain the same 1192 sequences but their names are GenBank accessions in the former and RefSeq accessions in the latter. Then you can convert the file to `.2bit` using a few lines of R code. See for example [this code](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/004/515/GCA_000004515.4_Glycine_max_v2.1/) which is the code that what used to accomplish this task for the `Amel_HAv3.1` assembly (**BSgenome.Amellifera.NCBI.AmelHAv3.1** package). Note that the code also takes care of renaming and sorting the sequences. Hopefully it won't be too hard to adapt the code to do the same thing for `Glycine_max_v2.1`.

Also please have a look at the seed file that was used to forge **BSgenome.Amellifera.NCBI.AmelHAv3.1**: https://github.com/Bioconductor/BSgenome/blob/master/inst/extdata/GentlemanLab/BSgenome.Amellifera.NCBI.AmelHAv3.1-seed
The seed file for `Glycine_max_v2.1` should be very similar.

Once **BSgenome.Gmax.NCBI.Gmv21** is ready, please submit it in a new issue.

Hope this helps.

H.

**Comment by Deisiany** [comment]
Hi Hervé,
No problem, this is the busiest time of the year for me as well.
Will do that, and will also use a 2.bit document for it.

Best,
Deisiany
Thanks, I'll check them out.

Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by Deisiany** [comment]
Hello Hervé,

I have tried a couple of things and for some reason *GenomeInfoDb* 1.31.4
version is not available, even deleting R, and Rstudio, and then installing
everything again, the latest version that I am getting
is GenomeInfoDb_1.28.4. which does not contain the soy sequences.
I have also removed and installed packages a couple of times as well as
force installed GenomeInfoDb with "BiocManager::install("GenomeInfoDb",
force = TRUE)".

Is there something that I am missing?

Thanks for your help!

Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by lshep** [comment]
What version of R are you installing?  What is your `sessionInfo()`    GenomeInfoDb 1.31.4 will only be available using R devel (R 4.2.0) and Bioconductor version 3.15.

**Comment by Deisiany** [comment]
I am on a Mac.

Here is my info:

> sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.2.1

Matrix products: default
LAPACK:
/Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8          BiocManager_1.30.16 compiler_4.1.2
 prettyunits_1.1.1   remotes_2.4.2
 [6] tools_4.1.2         testthat_3.1.2      digest_0.6.29
pkgbuild_1.3.1      pkgload_1.2.4
[11] evaluate_0.15       memoise_2.0.1       lifecycle_1.0.1
nlme_3.1-155        lattice_0.20-45
[16] rlang_1.0.2         cli_3.2.0           yaml_2.3.5
 parallel_4.1.2      xfun_0.30
[21] fastmap_1.1.0       withr_2.5.0         knitr_1.37          desc_1.4.1
         fs_1.5.2
[26] devtools_2.4.3      rprojroot_2.0.2     grid_4.1.2          glue_1.6.2
         R6_2.5.1
[31] processx_3.5.2      rmarkdown_2.13      sessioninfo_1.2.2
callr_3.7.0         purrr_0.3.4
[36] magrittr_2.0.2      ps_1.6.0            ellipsis_0.3.2
 htmltools_0.5.2     usethis_2.1.5
[41] ape_5.6-2           cachem_1.0.6        crayon_1.5.0
 brio_1.1.3


Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by lshep** [comment]
That version of R (4.1.2) is only available for Bioconductor 3.14.  The new assemblies were added to Bioconductor devel 3.15 and requires R-devel (R 4.2.0)

**Comment by Deisiany** [comment]
Thank you for the clarification.

I got R-devel working on my Mac. The problem is, now I have bioconductor
3.15 which seems to be unstable at the moment or at least that is how R
defines it.
Also, I cannot load GenomeInfoDb or a couple other packages such as
Biostrings under bioconductor. All of them kind of share the fact that
RCurl is missing. I tried to install RCurl in many different ways, but
unfortunately nothing worked.



*" > sessioninfo::package_info("GenomeInfoDb") ! package          * version
date (UTC) lib source   BiocGenerics     * 0.41.2  2021-11-19 [1]
Bioconductor   GenomeInfoDb       1.31.4  2022-01-30 [1] Bioconductor
 GenomeInfoDbData   1.2.7   2022-03-10 [1] Bioconductor   IRanges
 * 2.29.1  2021-11-19 [1] Bioconductor R RCurl              <NA>    <NA>
    [?] <NA>   S4Vectors        * 0.33.10 2022-01-12 [1] Bioconductor [1]
/Library/Frameworks/R.framework/Versions/4.2/Resources/library R ── Package
was removed from disk. "*

I believe the problem is that when trying to install GenomeInfoDb R looks
within 4.2 version and therefore it cannot find RCurl and in that way it
always fails loading even though I installed it RCurl on my Mac

*ERROR: loading failed*

** removing
‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RCurl’The
downloaded source packages are
in‘/private/var/folders/q8/xv5qyvd520v3lwlqnc2ffgvr0000gn/T/Rtmp87pTpY/downloaded_packages’Warning
messages:1: In .inet_warning(msg) :  unable to access index for repository
https://cran.rstudio.com/bin/macosx/contrib/4.2
<https://cran.rstudio.com/bin/macosx/contrib/4.2>:  cannot open URL
'https://cran.rstudio.com/bin/macosx/contrib/4.2/PACKAGES
<https://cran.rstudio.com/bin/macosx/contrib/4.2/PACKAGES>'2: In
.inet_warning(msg) :  installation of package ‘RCurl’ had non-zero exit
status*



Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by hpages** [comment]
Hi @Deisiany,

Very little information provided here to help you with the **RCurl** issue. In particular where did you get R devel from? How did you install **RCurl**? What error you got when trying to install it?

The easiest way to install R devel is to download it from https://mac.r-project.org/. Since you seem to have an Intel-based Mac, you should grab `R-devel.pkg` for high-sierra (even if you have Monterey).

To install **RCurl**:
```
if (!require("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
library(BiocManager)
valid()
```
Please show us the output of `valid()`.

Then:
```
install("RCurl")
```
Note that the CRAN folks are not providing Mac binary packages for R devel yet so you will be asked if you want to install **RCurl** from source. Say "yes". Be aware that you will need Xcode for this to work because **RCurl** contains C code that needs compilation.

Finally please try to improve the formatting/readability of your posts by placing code chunks (including their output) between triple backtick lines.

Thanks,
H.

**Comment by Deisiany** [comment]
Hi Hervé,
sorry, I included the error and apparently forgot to add the command.
I did download R-devel.pkg for high-sierra, and it was from the source you
provided above.

Here is my output:







*> valid()'getOption("repos")' replaces Bioconductor standard repositories,
see '?repositories' for detailsreplacement repositories:    CRAN:
https://cran.rstudio.com/ <https://cran.rstudio.com/>[1] TRUE*


Also, my output for when trying to install RCurl:











*> install("RCurl")'getOption("repos")' replaces Bioconductor standard
repositories, see '?repositories' for detailsreplacement repositories:
CRAN: https://cran.rstudio.com/ <https://cran.rstudio.com/>Bioconductor
version 3.15 (BiocManager 1.30.16), R Under development (unstable)
(2022-03-10 r81874)Installing package(s) 'RCurl'Package which is only
available in source form, and may need compilation of C/C++/Fortran:
‘RCurl’Do you want to attempt to install these from sources?
(Yes/no/cancel) yes*

..... # all looks fine after this
Then,



















*ld: warning: dylib (/usr/local/lib/liblzma.dylib) was built for newer
macOS version (11.0) than being linked (10.13)installing to
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/00LOCK-RCurl/00new/RCurl/libs**
R** data** inst** byte-compile and prepare package for lazy loading**
help*** installing help indices** building package indices** testing if
installed package can be loaded from temporary locationError: package or
namespace load failed for ‘RCurl’ in dyn.load(file, DLLpath = DLLpath,
...): unable to load shared object
'/Library/Frameworks/R.framework/Versions/4.2/Resources/library/00LOCK-RCurl/00new/RCurl/libs/RCurl.so':
dlopen(/Library/Frameworks/R.framework/Versions/4.2/Resources/library/00LOCK-RCurl/00new/RCurl/libs/RCurl.so,
0x0006): Library not loaded: @rpath/libcurl.4.dylib  Referenced from:
/Library/Frameworks/R.framework/Versions/4.2/Resources/library/00LOCK-RCurl/00new/RCurl/libs/RCurl.so
Reason: tried:
'/Library/Frameworks/R.framework/Resources/lib/libcurl.4.dylib' (no such
file),
'/Library/Java/JavaVirtualMachines/jdk1.8.0_241.jdk/Contents/Home/jre/lib/server/libcurl.4.dylib'
(no such file)Error: loading failedExecution haltedERROR: loading failed*
removing
‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/RCurl’*













*The downloaded source packages are in
‘/private/var/folders/q8/xv5qyvd520v3lwlqnc2ffgvr0000gn/T/RtmpiQTEkl/downloaded_packages’Warning
messages:1: In .inet_warning(msg) :  unable to access index for repository
https://cran.rstudio.com/bin/macosx/contrib/4.2
<https://cran.rstudio.com/bin/macosx/contrib/4.2>:  cannot open URL
'https://cran.rstudio.com/bin/macosx/contrib/4.2/PACKAGES
<https://cran.rstudio.com/bin/macosx/contrib/4.2/PACKAGES>'2: In
.inet_warning(msg) :  installation of package ‘RCurl’ had non-zero exit
statusThank you!!*


Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by Deisiany** [comment]
Hi,

I was able to find a way around this.
by going through the package pages and manually installing the RCurl
package.
Then I also needed to install other dependencies which were giving the
same error message.
For that I used the R-devel packages URL *https://cran.r-project.org/
<https://cran.r-project.org/> *
That way I could install all dependencies needed.
For example:
install.packages("rtracklayer", repos = "https://cran.r-project.org/",
dependencies = TRUE)

Thanks so much for all the help.

Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by Deisiany** [comment]
> Hi @Deisiany ,
> 
> Sorry for the slow response. I registered a few `Glycine max` NCBI assemblies in **GenomeInfoDb** 1.31.4:
> 
> ```
> library(GenomeInfoDb)
> 
> subset(registered_NCBI_assemblies(), organism=="Glycine max")
> #       organism         assembly       date               extra_info
> # 94 Glycine max Glycine_max_v2.1 2018/07/24     Cultivar:Williams 82
> # 95 Glycine max Glycine_max_v4.0 2021/03/10     Cultivar:Williams 82
> # 96 Glycine max     ASM2049715v1 2021/10/14       Cultivar:Hwangkeum
> # 97 Glycine max   GmBRS537.a1.v2 2021/10/15 Cultivar:EMBRAPA BRS 537
> #    assembly_accession circ_seqs
> # 94    GCF_000004515.5  MT, Pltd
> # 95    GCF_000004515.6  MT, Pltd
> # 96    GCA_020497155.1          
> # 97    GCA_012273815.2     
> 
> Seqinfo(genome="Glycine_max_v2.1")
> # Seqinfo object with 1192 sequences (2 circular) from Glycine_max_v2.1 genome:
> #   seqnames      seqlengths isCircular           genome
> #   Chr01           56831624      FALSE Glycine_max_v2.1
> #   Chr02           48577505      FALSE Glycine_max_v2.1
> #   Chr03           45779781      FALSE Glycine_max_v2.1
> #   Chr04           52389146      FALSE Glycine_max_v2.1
> #   Chr05           42234498      FALSE Glycine_max_v2.1
> #   ...                  ...        ...              ...
> #   scaffold_2573       3151      FALSE Glycine_max_v2.1
> #   scaffold_2654       2711      FALSE Glycine_max_v2.1
> #   scaffold_2706       3302      FALSE Glycine_max_v2.1
> #   scaffold_2709       9068      FALSE Glycine_max_v2.1
> #   scaffold_2735       2655      FALSE Glycine_max_v2.1
> ```
> 
> To forge **BSgenome.Gmax.NCBI.Gmv21** I strongly recommend that you use a `.2bit` file instead of a FASTA file. Converting a FASTA file to the `.2bit` format is not hard. You could download `GCF_000004515.5_Glycine_max_v2.1_genomic.fna.gz` from https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/515/GCF_000004515.5_Glycine_max_v2.1/, or `GCA_000004515.4_Glycine_max_v2.1_genomic.fna.gz` from https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/004/515/GCA_000004515.4_Glycine_max_v2.1/. Your choice. Both files should contain the same 1192 sequences but their names are GenBank accessions in the former and RefSeq accessions in the latter. Then you can convert the file to `.2bit` using a few lines of R code. See for example [this code](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/004/515/GCA_000004515.4_Glycine_max_v2.1/) which is the code that what used to accomplish this task for the `Amel_HAv3.1` assembly (**BSgenome.Amellifera.NCBI.AmelHAv3.1** package). Note that the code also takes care of renaming and sorting the sequences. Hopefully it won't be too hard to adapt the code to do the same thing for `Glycine_max_v2.1`.
> 
> Also please have a look at the seed file that was used to forge **BSgenome.Amellifera.NCBI.AmelHAv3.1**: https://github.com/Bioconductor/BSgenome/blob/master/inst/extdata/GentlemanLab/BSgenome.Amellifera.NCBI.AmelHAv3.1-seed The seed file for `Glycine_max_v2.1` should be very similar.
> 
> Once **BSgenome.Gmax.NCBI.Gmv21** is ready, please submit it in a new issue.
> 
> Hope this helps.
> 
> H.

Hi Hervé,
 Could you please send me the link to your .2bit?
 The link above leads me elsewhere.
 Also, I was able to get the 2.bit I am just not sure how it should look like so I would like to see an example to confirm if what I did is correct.
 
 Thank you!!

**Comment by hpages** [comment]
Hi @Deisiany,

Sorry for the delay.

I don't have a `.2bit` file for the `Glycine_max_v2.1` assembly, sorry. There must be some misunderstanding. In my earlier comment I was only trying to provide some basic guidance to help _you_ produce a `.2bit` file from the FASTA file available at NCBI for `Glycine_max_v2.1`. If you were able to produce this file, then great. If you were not, then please provide the details of what you did exactly and what went wrong.

To check your `.2bit` file, just try to load it in R:
```
library(rtracklayer)
import.2bit("path/to/the/2bit/file")
```
If that works, and if the returned DNAStringSet object looks ok (make sure that the names on it are as expected), then you can be confident that your `.2bit` file is correct.

BTW I see a file [here](https://github.com/Deisiany/BSgenome.Gmax.EnsemblPlants.Gmv2/tree/main/inst/extdata) (added to GitHub 17 months ago!) that has the `.2bit` extension but AFAICT it's not a valid `.2bit` file:
```
> import.2bit("single_sequences.2bit")
Error in .seqlengths_TwoBitFile(con) : UCSC library operation failed
In addition: Warning message:
In .seqlengths_TwoBitFile(con) :
  single_sequences.2bit doesn't have a valid twoBitSig
```

Regards,
H.

P.S.: You're still not formatting your posts properly by placing code chunks (including their output) between triple backtick lines. This makes the code chunks you're sharing with us very hard to read.

**Comment by hpages** [comment]
Hi @Deisiany, do you intend to follow up on this submission? Thanks.  -- H.

**Comment by Deisiany** [comment]
Yes I do,
I am still having trouble with the R-devel version where the sequences are
registered and that’s why I am taking so long to submit.
Once I figure out how to make that stable in my computer I will go ahead
with the submission.
Sorry for the delay!
Thank you for checking.


Deisiany Ferreira Neres (she/her/hers)
Ph.D. Student, Biological System Engineering
Virginia Polytechnic Institute and State University
HABB1, 1230 Washington St.
Blacksburg, VA 24060
(703) 598-5749

**Comment by hpages** [comment]
No worries. Thanks for the update.

H.

**Comment by Deisiany** [comment]
Hi @hpages,
I have package should work now. Sequences are in a .2bit format.

And I also altered the package name to BSgenome.Gmax.NCBI.Gmv40 since I am using the NCBI genome.

https://github.com/Deisiany/BSgenome.Gmax.NCBI.Gmv40.git

Please let me know if now works, or if there is anything else that needs to be done from my end.

**Comment by hpages** [comment]
Hi @Deisiany ,

Package looks good. Just a few minor issues:

- Running `R CMD check` on the source tarball obtained with `R CMD build` produces the following note:
    ```
    * checking top-level files ... NOTE
    File
      LICENSE
    is not mentioned in the DESCRIPTION file.
    ```
  This is because you've added a LICENSE file for Creative Commons Legal Code to the package. Note that this license conflicts with the Artistic-2.0 license. You need to choose one or the other. In any case if you keep this LICENSE file, it must be mentioned in the DESCRIPTION file with something like:
    ```
    License: <the name of the license> + file LICENSE
    ```
  Make sure that the changes you make are reflected on both the seed file and the DESCRIPTION file.

- You have the following  in the seed file and in the DESCRIPTION file:
    ```
    Description: Full genome sequences for Glycine max as provided by NCBI (assemblyGmv40, assembly accession GCF_000004515.5) and stored in Biostrings objects.
    ```
  Note that a space is missing in `assemblyGmv40`. Also the full name of the assembly is `Glycine_max_v4.0` so it should be `assembly Glycine_max_v4.0`. This would need to be corrected in both files (seed and DESCRIPTION).

Almost there.

Thanks,
H.

**Comment by Deisiany** [comment]
Hi @hpages,

I have made the alterations you have mentioned.

Thank you so much!

**Comment by hpages** [comment]
Thanks.

2 more things, sorry:

- As mentioned above, you need to add `+ file LICENSE` to your `License:` field. This is a requirement for any R package with a  LICENSE file in general.

- You have two `README.md` files (one in the top-level folder of your GitHub repo, and one inside the `BSgenome.Gmax.NCBI.Gmv40` package), and both of them contain the following:
    ```
    # Installing package
    
    Clone the repository locally, including the LFS of the repository, and  then install from the git clone, e.g., via `install.packages(“/path/to/local/clone”, repos = NULL, type = “Source”)`
    ```
  However, this is not how people should install a Bioconductor package. The recommended way to do this is with `BiocManager::install()`:
    ```
    if (!require("BiocManager", quietly=TRUE))
        install.packages("BiocManager")
    BiocManager::install("BSgenome.Gmax.NCBI.Gmv40")
    ```
    Of course this doesn't work at the moment but it will as soon as the package gets added to Bioconductor.

Thanks,
H.

**Comment by Deisiany** [comment]
Hi @hpages,

I am sorry!
I hope it correct this time. Let me know if I am mis-understanding what you are saying, but if I did understand now it is correct.

If it does not work I will switch back to License: Artistic-2.0 and that should make it easier, correct?

Thaank you!

**Comment by hpages** [comment]
Thanks for those corrections.

The DESCRIPTION file has:
```
Author: person("Deisiany", "Ferreira Neres", email = "deisianyufg@gmail.com", role = c("aut", "cre"))
Maintainer: Clay Wright <wright.clay@gmail.com>
```
This is incorrect. There are 2 ways to specify the authors/maintainer of a package: (1) with an `Author:` and `Maintainer:` field, OR (2) with an `Authors@R:` field. The 2 methods are mutually exclusive.

Note that the `person()` syntax can only be used in method (2) (i.e. in the `Authors@R:` field). It can not and should not be used in method (1). Please choose _one_ method and adjust as necessary.

Also it's confusing to see that Deisiany Ferreira Neres is listed as an author (`"aut"`) and a maintainer (`"cre"`) but then the `Maintainer:` field has a different person. If the two of you are going to be listed as maintainers, then you need to have:

    Author: Deisiany Ferreira Neres <deisianyufg@gmail.com>
    Maintainer: Deisiany Ferreira Neres <deisianyufg@gmail.com>, Clay Wright <wright.clay@gmail.com>

in the DESCRIPTION file, and the two of you will need to agree on the following:

> I am committed to the long-term maintenance of my package. This includes monitoring the support site for issues that users may have, subscribing to the bioc-devel mailing list to stay aware of developments in the Bioconductor community, responding promptly to requests for updates from the Core team in response to changes in R or underlying software.

Also make sure that the two email addresses in the DESCRIPTION file are valid. 

Thanks,
H.

**Comment by Deisiany** [comment]
Thank you!

I have done those and added the commitment agreement to the DESCRIPTION file.
Both emails are correct!

Thanks,
Deisiany

**Comment by hpages** [comment]
Please do not add the commitment to the DESCRIPTION file. Because of this, the package can no longer be built:
```
hpages@spectre:~$ R CMD build BSgenome.Gmax.NCBI.Gmv40
* checking for file ‘BSgenome.Gmax.NCBI.Gmv40/DESCRIPTION’ ...Error : file '/home/hpages/pkgreviews/BSgenome.Gmax.NCBI.Gmv40.review/BSgenome.Gmax.NCBI.Gmv40/BSgenome.Gmax.NCBI.Gmv40/DESCRIPTION' is not in valid DCF format
 EXISTS but not correct format
```
You just need to say here that you commit to the long-term maintenance of your package, that's all. Thanks!

**Comment by Deisiany** [comment]
Sorry,
I removed it.

**Comment by hpages** [comment]
Thanks. I just added the package to Bioconductor. It will become available via `BiocManager::install()` in the next 3 hours or so.

I'm closing this issue now.

Cheers,
H.
